### PR TITLE
chore(sdk): add docstrings to `used_artifacts` and `logged_artifacts`

### PR DESCRIPTION
**Pull request by trane293**
Description
-----------

- Fixes [WB-21539](https://wandb.atlassian.net/browse/WB-21539)

What does the PR do? Include a concise description of the PR contents.

Adds missing docstrings to two artifact APIs. 

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

[WB-21539]: https://wandb.atlassian.net/browse/WB-21539?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ibindlish — APPROVED** [review]

File: wandb/apis/public/runs.py [hunk 1]
```diff
@@ -754,10 +754,54 @@ def scan_history(self, keys=None, page_size=1000, min_step=None, max_step=None):
 
     @normalize_exceptions
     def logged_artifacts(self, per_page=100):
+        """Return an iterable collection of all artifacts logged by this run.
+
+        This method retrieves all output artifacts that were logged during the run.
+
+        Arguments:
```
Comment (kptkin) [inline wandb/apis/public/runs.py:761]:
```suggestion
        Args:
```

File: wandb/apis/public/runs.py [hunk 2]
```diff
@@ -754,10 +754,54 @@ def scan_history(self, keys=None, page_size=1000, min_step=None, max_step=None):
...
+        Example:
+            ```python
+            run = api.run("entity/project/run_id")
+            for artifact in run.logged_artifacts():
+                print(f"{artifact.name}:{artifact.version}")
+            ```
+
+        Note:
+            This method returns a paginated result. You can iterate over all artifacts
+            or use list() to collect them into a single list.
```
Comment (kptkin) [inline wandb/apis/public/runs.py:776]: we follow the google docstring style

File: wandb/apis/public/runs.py [hunk 3]
```diff
@@ -754,10 +754,54 @@ def scan_history(self, keys=None, page_size=1000, min_step=None, max_step=None):
...
+            RunArtifacts: An iterable collection of all Artifact objects logged as outputs during this run.
+
+        Example:
+            ```python
+            run = api.run("entity/project/run_id")
+            for artifact in run.logged_artifacts():
+                print(f"{artifact.name}:{artifact.version}")
+            ```
+
+        Note:
```
Comment (kptkin) [inline wandb/apis/public/runs.py:774]: note before args

File: wandb/apis/public/runs.py (hunk 2 above)
Comment (kptkin) [reply wandb/apis/public/runs.py:776]: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings

File: wandb/apis/public/runs.py [hunk 4]
```diff
@@ -754,10 +754,54 @@ def scan_history(self, keys=None, page_size=1000, min_step=None, max_step=None):
...
+            This method returns a paginated result. You can iterate over all artifacts
+            or use list() to collect them into a single list.
+        """
         return public.RunArtifacts(self.client, self, mode="logged", per_page=per_page)
 
     @normalize_exceptions
     def used_artifacts(self, per_page=100):
```
Comment (kptkin) [inline wandb/apis/public/runs.py:781]: if you already here mind adding type annotation

File: wandb/apis/public/runs.py (hunk 1 above)
Comment (trane293) [reply wandb/apis/public/runs.py:761]: @kptkin The `Run` class functions all use the word "Arguments", hence the usage here. Changing it to `Args` will make this inconsistent from the whole file.

Comment (kptkin) [reply wandb/apis/public/runs.py:761]: that's fine, we want to align to the google docstring style, so let's do the correct ones for the new ones we are adding

File: wandb/apis/public/runs.py (hunk 3 above)
Comment (trane293) [reply wandb/apis/public/runs.py:774]: Done

File: wandb/apis/public/runs.py (hunk 2 above)
Comment (trane293) [reply wandb/apis/public/runs.py:776]: Noted. Unsure what's the action item here though.

Comment (kptkin) [reply wandb/apis/public/runs.py:776]: same as above using Args (linked the guide for more details)

File: wandb/apis/public/runs.py (hunk 4 above)
Comment (trane293) [reply wandb/apis/public/runs.py:781]: Done, added type annotation to the two functions currently in question. Added a ticket to backlog to revisit and check the Run class for any extra type annotations needed.

File: wandb/apis/public/runs.py (hunk 2 above)
Comment (trane293) [reply wandb/apis/public/runs.py]: Changed the docstrings to match Google style.

File: wandb/apis/public/runs.py [hunk 5]
```diff
@@ -753,11 +753,51 @@ def scan_history(self, keys=None, page_size=1000, min_step=None, max_step=None):
             )
 
     @normalize_exceptions
-    def logged_artifacts(self, per_page=100):
+    def logged_artifacts(self, per_page: int = 100) -> public.RunArtifacts:
+        """Fetches all artifacts logged by this run.
+
+        Retrieves all output artifacts that were logged during the run.
+
+        Note:
```
Comment (kptkin) [inline wandb/apis/public/runs.py:761]: do we really need this note? maybe just add it to the rest of the above text?

File: wandb/apis/public/runs.py [hunk 6]
```diff
@@ -753,11 +753,51 @@ def scan_history(self, keys=None, page_size=1000, min_step=None, max_step=None):
...
+        Args:
+            per_page: Number of artifacts to fetch per API request. Defaults to 100.
+
+        Returns:
+            An iterable collection of all Artifact objects logged as outputs during this run.
+
+        Example:
+            run = api.run("entity/project/run_id")
+            for artifact in run.logged_artifacts():
+                print(f"{artifact.name}:{artifact.version}")
```
Comment (kptkin) [inline wandb/apis/public/runs.py:774]: could you change it to use doctest format?

**Review by kptkin — APPROVED** [review]

**Review by kptkin — APPROVED** [review]

**Pull request merged**